web/api/v1/api_test.go

```diff
@@ -773,6 +775,13 @@ func TestLabelNames(t *testing.T) {
 			expected: []string{"__name__", "abc", "foo", "xyz"},
 			api:      api,
 		},
+		{
+			name:     "non empty label matcher with limit",
+			matchers: []string{`{foo=~".+"}`},
+			expected: []string{"__name__", "abc"},
```

Comment:
Is it guaranteed that they'll always be the same?

Reply:
IIUC, the response from the querier is sorted. So the first two entries should always be the same? Please let me know if that's not the case.

I also ran the test 100K times just to check if it always passes.

Reply:
I wasn't sure, so looked it up: https://github.com/prometheus/prometheus/blob/f5eff89811cb3249f5fb3345e3a2350936b0362c/tsdb/block.go#L98
https://github.com/prometheus/prometheus/blob/f5eff89811cb3249f5fb3345e3a2350936b0362c/storage/interface.go#L166

Reply:
Yes, I'm a lazybones :) Thanks to you two.